src/core/mesh/qgsmeshlayertemporalproperties.cpp

```diff
@@ -0,0 +1,82 @@
+#include "qgsmeshlayertemporalproperties.h"
+#include "qgsmeshdataprovidertemporalcapabilities.h"
+#include "qgsproject.h"
+#include "qgsprojecttimesettings.h"
+
+QgsMeshLayerTemporalProperties::QgsMeshLayerTemporalProperties( QObject *parent, bool enabled ): QgsMapLayerTemporalProperties( parent, enabled )
+{}
+
+QDomElement QgsMeshLayerTemporalProperties::writeXml( QDomElement &element, QDomDocument &doc, const QgsReadWriteContext &context )
+{
+  Q_UNUSED( context );
+
+  QDomElement temporalElement = doc.createElement( QStringLiteral( "temporal" ) );
+  temporalElement.setAttribute( QStringLiteral( "temporal-active" ), isActive() ? true : false );
+  temporalElement.setAttribute( QStringLiteral( "reference-time" ), mReferenceTime.toTimeSpec( Qt::UTC ).toString( Qt::ISODate ) );
+  temporalElement.setAttribute( QStringLiteral( "start-time-extent" ), mTimeExtent.begin().toTimeSpec( Qt::UTC ).toString( Qt::ISODate ) );
+  temporalElement.setAttribute( QStringLiteral( "end-time-extent" ), mTimeExtent.end().toTimeSpec( Qt::UTC ).toString( Qt::ISODate ) );
+
+  element.appendChild( temporalElement );
+
+  return element;
+}
+
+bool QgsMeshLayerTemporalProperties::readXml( const QDomElement &element, const QgsReadWriteContext &context )
+{
+  Q_UNUSED( context );
+
+  QDomElement temporalElement = element.firstChildElement( QStringLiteral( "temporal" ) );
+  bool active = temporalElement.attribute( QStringLiteral( "temporal-active" ) ).toInt();
+  setIsActive( active );
+
+  mReferenceTime = QDateTime::fromString( temporalElement.attribute( QStringLiteral( "reference-time" ) ), Qt::ISODate );
+
+  if ( temporalElement.hasAttribute( QStringLiteral( "start-time-extent" ) )
+       && temporalElement.hasAttribute( QStringLiteral( "end-time-extent" ) ) )
+  {
+    QDateTime start = QDateTime::fromString( temporalElement.attribute( QStringLiteral( "start-time-extent" ) ), Qt::ISODate );
+    QDateTime end = QDateTime::fromString( temporalElement.attribute( QStringLiteral( "end-time-extent" ) ), Qt::ISODate );
+    mTimeExtent = QgsDateTimeRange( start, end );
+  }
+
+  return true;
+}
+
+void QgsMeshLayerTemporalProperties::setDefaultsFromDataProviderTemporalCapabilities( const QgsDataProviderTemporalCapabilities *capabilities )
+{
+  const QgsMeshDataProviderTemporalCapabilities *tempCap = static_cast<const QgsMeshDataProviderTemporalCapabilities *>( capabilities );
+
+  setIsActive( tempCap->hasTemporalCapabilities() );
+  mReferenceTime = tempCap->referenceTime();
+
+  if ( mReferenceTime.isValid() )
+    mTimeExtent = tempCap->timeExtent();
+  else // If the provider capabilittyies doesn't provide a reference time, try to bring one frome projet settings
+  {
+    // Not sure it is a good idea to call instance of project here ...
+    mReferenceTime = QgsProject::instance()->timeSettings()->temporalRange().begin();
```

Comment:
I think the reference time could be member variable of the QgsMeshDataProviderTemporalCapabilities, and changed by setter when QgsProject::instance()->timeSettings()->temporalRange() is changed or manually changed. probably not good to call instance() here

Reply:
I think the principle of those class is :
- capabilities : what can the provider handle from the data, attribute of the class are only set from the provider and from data, as much as possible (provider time unit is a breach in the rule, but it is a provider capability that needs to be used in the capabilities class).
- properties : settings from other part, that is user or project settings.